spec.bs

```diff
@@ -347,6 +530,196 @@ Moreover, each {{SharedStorageWorklet}}'s [=global scopes|list of global scopes=
+  </div>
+
+  <div class="example">
+    The [=default index=] could be defined to be [=selectURL/input URL list=]'s [=list/size=] minus 1.
+
+    In this case, whenever the {{SharedStorageSelectURLOperation}}'s "`run`" method encounters an error, or whenever there is insufficient [=calling site/remaining navigation budget=], {{SharedStorageWorklet/selectURL()}} would return a {{SharedStorageResponse}} for the last {{SharedStorageUrlWithMetadata/url}} in its [=selectURL/input URL list=].
+  </div>
+
+  <div algorithm>
+    To <dfn>determine remaining navigation budget</dfn>, given an [=environment settings object=] |environment| and a [=calling site=] |site|, run the following steps:
+
+    1. If |site| is an [=opaque origin=], return undefined.
+    1. Let |maxBits| be the [=user agent=]'s [=navigation entropy allowance=].
+    1. If the [=user agent=]'s [=shared storage navigation budget table=] does not [=map/contain=] |site|, then return |maxBits|.
+    1. Otherwise, let |ledger| be [=user agent=]'s [=shared storage navigation budget table=][|site|].
+    1. Let |debitSum| be 0.
+    1. [=map/iterate|For each=] [=list/item=] |bitDebit| in |ledger|, do the following steps:
+        1. Let |debit| be |bitDebit|'s [=bit debit/bits=].
+        1. If the result of running [=check whether a bit debit is expired=] with |environment| and |bitDebit| is false, then increment |debitSum| by |debit|.
+    1. Return |maxBits| minus |debitSum|.
+  </div>
+
+  <div algorithm>
+    To <dfn>check whether a bit debit is expired</dfn>, given an [=environment settings object=] |environment| and a [=bit debit=] |bitDebit|, run the following steps:
+
+    1. Let |epochLength| be the [=user agent=]'s [=navigation budget lifetime=].
+    1. Let |currentTime| be |environment|'s [=environment settings object/current wall time=].
+    1. Let |threshold| be |currentTime| minus |epochLength|.
+    1. If |bitDebit|'s [=bit debit/timestamp=] is less than |threshold|, return true.
+    1. Otherwise, return false.
+  </div>
+
+  A [=bit debit=] will need to be [=charge shared storage navigation budget|charged=] to the [=shared storage navigation budget table=] for each [=top-frame=] [=navigate|navigation=] initiated by a [=fenced frame=] whose {{FencedFrameConfig}} was generated via {{SharedStorageWorklet/selectURL()}}, as this can leak cross-site data. Since the [=bit debit/bits=] to charge is calculated during the call to {{SharedStorageWorklet/selectURL()}} but only actually recorded in the [=shared storage navigation budget table=] if and when the resulting fenced frame initiates a [=top-frame=] [=beginning navigation|navigation=], the [=bit debit/bits=] must be stored as a <dfn>pending shared storage budget debit</dfn> in the corresponding {{FencedFrameConfig}} until this time.
```

Comment:
```suggestion
  A [=bit debit=] will need to be [=charge shared storage navigation budget|charged=] to the [=shared storage navigation budget table=] for each [=top-frame=] [=navigate|navigation=] initiated by a [=fenced frame=] whose {{FencedFrameConfig}} was generated via {{SharedStorageWorklet/selectURL()}}, as this can leak cross-site data. Since the [=bit debit/bits=] to charge is calculated during the call to {{SharedStorageWorklet/selectURL()}} but only actually recorded in the [=shared storage navigation budget table=] if and when the resulting fenced frame initiates a [=top-level traversable=] [=beginning navigation|navigation=], the [=bit debit/bits=] must be stored as a <dfn>pending shared storage budget debit</dfn> in the corresponding {{FencedFrameConfig}} until this time.
```